external_search.py

```diff
@@ -621,9 +651,12 @@ def bulk_update(self, works, retry_on_batch_failure=True):
         )
 
         # If the entire update failed, try it one more time before giving up on the batch.
-        if retry_on_batch_failure and len(errors) == len(docs):
-            self.log.info("Elasticsearch bulk update timed out, trying again.")
-            return self.bulk_update(works, retry_on_batch_failure=False)
+        if len(errors) == len(docs):
+            if retry_on_batch_failure:
+                self.log.info("Elasticsearch bulk update timed out, trying again.")
+                return self.bulk_update(works, retry_on_batch_failure=False)
+            else:
+                docs = []
```

Comment:
Previously, if every document in the update failed for reasons other than a timeout (or if a timeout happened twice), each document was treated both as a success and an error.